File: testers/avro/java-spring/generated-and-checked-in/com/example/events/LinkedListNodePublisher.java
package com.example.events;

import java.util.concurrent.CompletableFuture;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.kafka.support.SendResult;
import org.springframework.stereotype.Service;

/** Type-safe event publisher for linked-list-node topic */
@Service
public record LinkedListNodePublisher(
    KafkaTemplate<String, LinkedListNode> kafkaTemplate, String topic) {
  public LinkedListNodePublisher(KafkaTemplate<String, LinkedListNode> kafkaTemplate) {
    this(kafkaTemplate, "linked-list-node");
  }

  public LinkedListNodePublisher withKafkaTemplate(
      KafkaTemplate<String, LinkedListNode> kafkaTemplate) {
    return new LinkedListNodePublisher(kafkaTemplate, topic);
  }

  public LinkedListNodePublisher withTopic(String topic) {
    return new LinkedListNodePublisher(kafkaTemplate, topic);
  }

  /** Publish a LinkedListNode event */
  public CompletableFuture<SendResult<String, LinkedListNode>> publish(
      String key, LinkedListNode event) {
    return kafkaTemplate.send(topic, key, event);
  }
}

File: testers/avro/scala-cats/generated-and-checked-in/com/example/events/LinkedListNodePublisher.scala
package com.example.events

import cats.effect.IO
import fs2.kafka.KafkaProducer
import fs2.kafka.ProducerRecord
import fs2.kafka.ProducerRecords
import fs2.kafka.ProducerResult
import scala.annotation.unused

@unused
/** Type-safe event publisher for linked-list-node topic */
case class LinkedListNodePublisher(
  kafkaTemplate: KafkaProducer[IO, String, LinkedListNode],
  topic: String = "linked-list-node"
) {
  /** Publish a LinkedListNode event */
  def publish(
    key: String,
    event: LinkedListNode
  ): IO[ProducerResult[String, LinkedListNode]] = {
    return kafkaTemplate.produce(ProducerRecords.one(ProducerRecord(topic, key, event))).flatten
  }
}